src/main/java/uk/nhs/ctp/service/ValidationService.java

```diff
@@ -6,54 +6,72 @@
 import java.nio.charset.StandardCharsets;
 import java.nio.file.attribute.FileTime;
 import java.util.HashSet;
-import java.util.stream.Collectors;
+import java.util.List;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
-import uk.nhs.ctp.audit.model.AuditEntry;
-import uk.nhs.ctp.auditFinder.finder.AuditFinder;
+import uk.nhs.ctp.audit.model.AuditSession;
+import uk.nhs.ctp.auditFinder.model.OperationType;
 
 @Service
 @RequiredArgsConstructor
 public class ValidationService {
 
-  private final AuditFinder auditFinder;
+  private static final String CASE_ID = "caseId";
 
-  public byte[] zipResources(Long caseId) throws IOException {
-    var audits = auditFinder.findAllEmsEncountersByCaseId(caseId.toString());
-    var auditEntries = audits.stream()
-        .flatMap(a -> a.getEntries().stream())
-        .collect(Collectors.toUnmodifiableList());
+  public byte[] zipAudits(List<AuditSession> audits, OperationType operationType) throws IOException {
+    switch (operationType) {
+      case ENCOUNTER:
+        if (audits.stream().anyMatch(a -> !a.getAdditionalProperties().containsKey(CASE_ID))) {
+          throw new UnsupportedOperationException("Encounter audits must have caseId set");
+        }
+        break;
+      case SERVICE_SEARCH:
+        if (audits.size() != 1 || audits.get(0).getEntries().size() != 1) {
+          throw new UnsupportedOperationException(
+              "Can only zip service_search audits one at a time");
+        }
+        break;
+      default:
+        throw new UnsupportedOperationException("Non-standard audit operation types not supported");
+    }
 
     var output = new ByteArrayOutputStream();
     var zip = new ZipOutputStream(output);
 
     var bundledResources = new HashSet<String>();
 
-    for (AuditEntry entry : auditEntries) {
-      if (!entry.getRequestMethod().equals("GET")) {
-        continue;
-      }
+    for (var audit : audits) {
```

Comment:
For audits captured by the CDSS we will definitely need to zip the request/response of the outer audit record, not just the entries